### ISSUE TITLE
No display on Vive

### ISSUE THREAD

**Issue by yaomtc** [ValveSoftware/SteamVR-for-Linux#108]
#### Your system information
* Steam client version (build number or date): May 18 2018
* Distribution (e.g. Ubuntu): Kubuntu 18.04
* Graphics driver version (run nvidia-settings): Mesa 18.0.0
* Gist for SteamVR System Information: https://gist.github.com/yaomtc/d46710d59fa120cb43e013f8ebebce5f
* Opted into Steam client beta?: Yes
* Opted into SteamVR beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
After having installed Kubuntu today, and installing the Vulkan Mesa drivers and following the other instructions here, I managed to get the Vive completely working except the most important part: the display. Motion works, controller works, setup works, what I'm seeing in the compositor window (that I can't move to the Vive display) works great. ~~It seems to me that my AMD card not supporting direct mode, and the latest -buntus using Linux 4.15, could be the problem:~~

https://www.phoronix.com/scan.php?page=news_item&px=Linux-4.15-Non-Desktop-Quirk

EDIT: [This person](https://github.com/ValveSoftware/SteamVR-for-Linux/issues/102) got their Vive working with Linux 4.15 on Debian Testing a month ago. Not sure what they have that I need, I tried moving that ICD file too.

#### Steps for reproducing this issue:
1. Have AMD graphics card and Kubuntu 18.04
2. Install and launch SteamVR
3. Look into the Vive headset and see nothing?

**Comment by ChristophHaag** [comment]
I don't know about ubuntu 18.04 but if it has a new enough software stack, specifically xorgproto/X.org, it may already be hiding the HMD with non-desktop. However, direct mode does not work yet without patching mesa/radv and xf86-video-amdgu or using modesetting ddx.

I have some more info about the current status at https://haagch.frickel.club/#!drmlease.md

tl;dr: If you run `xrandr` and see the HMD as `disconnected` but having modes, it is being hidden from X due to the non-desktop property. In this case you can termporarily enable extended mode with `xrandr --output HDMI-A-0 --set non-desktop 0`, subsituting the output name where your Vive is of course.

**Comment by yaomtc** [comment]
I found out it was the DisplayPort cable I was using. Switched to a DVI-D to HDMI cable for my monitor, and the Vive's included HDMI cable to connect to the link box. Now the Vive displays things. Will test more tonight. Thanks for the info!
